src/main/java/be/seeseemelk/mockbukkit/potion/MockInternalPotionData.java

```diff
@@ -45,6 +47,9 @@ public MockInternalPotionData(NamespacedKey namespacedKey)
 		this.potionEffects = tempPotionEffects;
 		this.upgradeable = Registry.POTION.get(new NamespacedKey(namespacedKey.getNamespace(), "strong_" + namespacedKey.getKey())) != null;
 		this.extendable = Registry.POTION.get(new NamespacedKey(namespacedKey.getNamespace(), "long_" + namespacedKey.getKey())) != null;
+		this.upgraded = namespacedKey.getKey().startsWith("strong_");
+		this.extended = namespacedKey.getKey().startsWith("long_");
```

Comment:
I need these two flags so I could implement the methods in `PotionMetaMock`.

Reply:
Someone else tried / is trying to do something like this, convert between potion data and potion effect type. See comment https://github.com/MockBukkit/MockBukkit/pull/1023#discussion_r1642457994 on how to do so properly.

Most notably the namespaced key supplied into the internal potion data constructor should be from a base potion effect. You can see that that is assumed in the two lines above (I think I wrote that code...).

Reply:
So is what I wrote fine? I dont mind if it changed later. For now I just want the tests to pass where I access it to test something else.

Reply:
I do not like this implementation, as it adds functionallity to the MockInternalPotionData which it is not meant to have. Even if you say it's temporary, this is often not the case in software development. This is also getting out of scope

Even though it's not ideal, can't you just avoid this in the tests for now? For say, not use upgraded / extended potion effects.

I hope that clears up stuff / I'm trying to be more clear when reviewing, but I still have a lot to learn

Reply:
I can disable the tests for now. I cannot just avoid these in tests. I cannot create any potion itemstack, because setting the base potion type throws an NotImplementedError.

Reply:
I see, but you can't set the potion data instead of potion effect type?

Reply:
Line 371 in `PlayerMock` is `PotionType.InternalPotionData internalPotionData = server.getUnsafe().getInternalPotionData(Registry.POTION.getKey(potionMeta.getBasePotionType()));`. The `potionMeta.getBasePotionType()` throws the error (I accidentally left it implemented, changing it to throwing an error right now). I don't see any way around this currently without implementing it. I'm not fully aware what the problem is and why can't can be fixed easily with another few parameters to store data. Essentially only a class variable to store and return what is set and accessed in the basePotionType methods?